### PR TITLE
Use `max_code_size` and `max_wasm_data_size` from Parachains Configuration

### PR DESCRIPTION
**Pull request by shawntabrizi**
This updates the Registrar pallet to use the `max_code_size` and `max_wasm_data_size` from the Configuration pallet, rather than hardcoded constants which are not in sync with the on-chain configuration.

Note there is still a hardcoded constant in `primitives_v1`:

```
/// Maximum compressed code size we support right now.
/// At the moment we have runtime upgrade on chain, which restricts scalability severely. If we want
/// to have bigger values, we should fix that first.
pub const MAX_CODE_SIZE: u32 = 3 * 1024 * 1024;
```

This seems to be used in the chain specification and `impl Protocol`. cc @eskimor

### CODE REVIEW

**Review by rphmeier — APPROVED** [review]

**Comment by eskimor** [comment]
The question is whether it makes sense to let things be configurable by the runtime, if it is actually a limitation of the node, like `MAX_CODE_SIZE`. I mean it is obviously fine to set it to a lower value, but the constants should be an upper limit to what a user can choose in the runtime configuration. If we became able to increase that limit at some point, the runtime would get the new limit via the next runtime upgrade.
Knowing those upper limits at compile time is pretty useful, I would not like to abandon that.

**Review by eskimor — APPROVED** [review]

File: runtime/kusama/src/lib.rs [hunk 1]
```diff
@@ -1105,9 +1105,7 @@ impl parachains_initializer::Config for Runtime {
 }
 
 parameter_types! {
-	pub const ParaDeposit: Balance = deposit(10, MAX_CODE_SIZE + MAX_HEAD_SIZE);
-	pub const MaxCodeSize: u32 = MAX_CODE_SIZE;
-	pub const MaxHeadSize: u32 = MAX_HEAD_SIZE;
+	pub const ParaDeposit: Balance = 40 * UNITS;
```
Comment (ordian) [inline runtime/kusama/src/lib.rs:1108]: FWIW this was previously ~11.2 UNITS

Comment (shawntabrizi) [reply runtime/kusama/src/lib.rs:1108]: It was 35 KSM from an on-chain query

File: node/primitives/src/lib.rs [hunk 2]
```diff
@@ -41,21 +41,18 @@ use polkadot_primitives::v1::{
...
 
 pub mod approval;
 
-/// The bomb limit for decompressing code blobs.
-pub const VALIDATION_CODE_BOMB_LIMIT: usize = 16 * 1024 * 1024;
-
-/// Maximum PoV size we support right now.
-pub const MAX_POV_SIZE: u32 = 20 * 1024 * 1024;
+/// The bomb limit for decompressing code blobs. (8x growth)
+pub const VALIDATION_CODE_BOMB_LIMIT: usize = (MAX_CODE_SIZE * 8u32) as usize;
```
Comment (shawntabrizi) [inline node/primitives/src/lib.rs:52]:
```suggestion
/// The bomb limit for decompressing code blobs.
pub const VALIDATION_CODE_BOMB_LIMIT: usize = (MAX_CODE_SIZE * 4u32) as usize;
```

**Review by rphmeier — APPROVED** [review]

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**